File: android/app/src/main/java/com/djs66256/short_drama/data/dto/BookingAssetDto.kt
package com.djs66256.short_drama.data.dto

import com.djs66256.short_drama.domain.model.BookingAsset
import com.djs66256.short_drama.domain.model.BookingAssetStatus
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class BookingAssetDto(
    @SerialName("drama_id")
    val dramaId: String,
    val title: String,
    @SerialName("cover_url")
    val coverUrl: String? = null,
    @SerialName("episode_count")
    val episodeCount: Int,
    @SerialName("booked_at")
    val bookedAt: String,
    @SerialName("availability_status")
    val availabilityStatus: String,
) {
    fun toDomain(): BookingAsset {
        val status = requireNotNull(BookingAssetStatus.fromApiValue(availabilityStatus)) {
            "Unsupported booking asset status: $availabilityStatus"
        }
        return BookingAsset(
            dramaId = dramaId,
            title = title,
            coverUrl = coverUrl.orEmpty(),
            episodeCount = episodeCount,
            bookedAt = bookedAt,
            availabilityStatus = status,
        )
    }
}

File: ios/ShortDrama/Sources/Data/DTOs/BookingAssetDTO.swift
import Foundation

struct BookingAssetDTO: Codable, Equatable {
    let dramaID: String
    let title: String
    let coverURL: String?
    let episodeCount: Int
    let bookedAt: String
    let availabilityStatus: BookingAssetAvailabilityStatus

    private enum CodingKeys: String, CodingKey {
        case dramaID = "dramaId"
        case title
        case coverURL = "coverUrl"
        case episodeCount
        case bookedAt
        case availabilityStatus
    }
}

extension BookingAssetDTO {
    func toEntity() -> BookingAsset {
        BookingAsset(
            dramaID: dramaID,
            title: title,
            coverURL: coverURL,
            episodeCount: episodeCount,
            bookedAt: bookedAt,
            availabilityStatus: availabilityStatus
        )
    }
}
